File: Python/ExampleTests/production/OrderItemCollection.py
from .Dollar import Dollar
from .OrderItem import OrderItem

ORDER_ITEM_COLLECTION_MINIMUM = 0
ORDER_ITEM_COLLECTION_MAXIMUM = 100


class OrderItemCollection:
    def __init__(self):
        self._items: list[OrderItem] = []

    def add(self, item: OrderItem) -> None:
        self._items.append(item)

    def read(self) -> list[OrderItem]:
        return list(self._items)

    def size(self) -> int:
        return len(self._items)

    def compute_total(self) -> Dollar:
        total = Dollar('0')
        for item in self._items:
            total = total.plus(item.item_total)
        return total

    def __eq__(self, other):
        if not isinstance(other, OrderItemCollection):
            return NotImplemented
        return self._items == other._items

    def __str__(self):
        return f'OrderItemCollection{{{len(self._items)} items}}'

File: Rust/ExampleTests/production/orderitemcollection.rs
use super::*;

pub const ORDER_ITEM_COLLECTION_MINIMUM: usize = 0;
pub const ORDER_ITEM_COLLECTION_MAXIMUM: usize = 100;

#[derive(Debug, Clone, Default, PartialEq)]
pub struct OrderItemCollection {
    items: Vec<OrderItem>,
}

impl OrderItemCollection {
    pub fn new() -> Self { Self::default() }

    pub fn add(&mut self, item: OrderItem) { self.items.push(item); }
    pub fn read(&self) -> &[OrderItem] { &self.items }
    pub fn size(&self) -> usize { self.items.len() }

    pub fn compute_total(&self) -> Dollar {
        self.items.iter().fold(Dollar::default(), |acc, i| acc.plus(&i.item_total))
    }
}
